src/sqlfluff/core/rules/functional/segments.py

```diff
@@ -64,6 +64,8 @@ def raw_slices(self) -> RawFileSlices:
             )
         raw_slices = set()
         for s in self:
+            if s.pos_marker is None:
+                raise ValueError("Segments include a positionless segement")
```

Comment:
To make the coverage check happy.
```suggestion
                raise ValueError("Segments include a positionless segement")  # pragma: no cover
```